crates/client/src/client/schema/blob.rs

```diff
@@ -26,3 +26,20 @@ pub struct Blob {
     pub id: BlobId,
     pub bytecode: HexString,
 }
+
+#[derive(cynic::QueryFragment, Clone, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl", graphql_type = "Blob")]
+pub struct BlobIdFragment {
+    pub id: BlobId,
+}
+
+#[derive(cynic::QueryFragment, Clone, Debug)]
+#[cynic(
+    schema_path = "./assets/schema.sdl",
+    graphql_type = "Query",
+    variables = "BlobByIdArgs"
+)]
+pub struct BlobExistsQuery {
+    #[arguments(id: $id)]
+    pub blob: Option<BlobIdFragment>,
```

Comment:
is there a reason we don't just return a bool here instead of returning the fragment?
we also need to dimension the query cost for `blob_exists`, as I can see we do it for the `bytecode`: https://github.com/FuelLabs/fuel-core/blob/fa1a049fd0cdde7bc7da6f2164d4a9aef8103d49/crates/fuel-core/src/schema/blob.rs#L31

Reply:
Complexity added in #2299

Reply:
noticed!